cirq/experiments/fidelity_estimation.py

```diff
@@ -496,32 +556,97 @@ def sample_2q_xeb_circuits(
+        A pandas dataframe with index given by ['circuit_i', 'cycle_depth'].
+        Columns always include "sampled_probs". If `combinations_by_layer` is
+        not `None` and you are doing parallel XEB, additional metadata columns
+        will be attached to the returned DataFrame.
     """
+    # Set up progress reporting
     if progress_bar is None:
         progress_bar = _NoProgress
 
-    q0, q1 = _verify_and_get_two_qubits_from_circuits(circuits)
+    # Shim isolated-XEB as a special case of combination-style parallel XEB.
+    if combinations_by_layer is None:
+        q0, q1 = _verify_and_get_two_qubits_from_circuits(circuits)
+        circuits = [
+            circuit.transform_qubits(
+                lambda q: {q0: devices.LineQubit(0), q1: devices.LineQubit(1)}[q]
+            )
+            for circuit in circuits
+        ]
+        combinations_by_layer = [
+            CircuitLibraryCombination(
+                layer=None,
+                combinations=np.arange(len(circuits))[:, np.newaxis],
+                pairs=[(q0, q1)],
+            )
+        ]
+        one_pair = True
+    else:
+        _verify_two_line_qubits_from_circuits(circuits)
+        one_pair = False
+
+    # Check `combinations_by_layer` is compatible with `circuits`.
+    for layer_combinations in combinations_by_layer:
+        if np.any(layer_combinations.combinations < 0) or np.any(
+            layer_combinations.combinations >= len(circuits)
+        ):
+            raise ValueError("`combinations_by_layer` has invalid indices.")
+
+    # Construct fully-wide "zipped" circuits.
+    zipped_circuits: List[_ZippedCircuit] = []
+    for layer_i, layer_combinations in enumerate(combinations_by_layer):
+        for combination_i, combination in enumerate(layer_combinations.combinations):
+            wide_circuit = Circuit.zip(
+                *(
+                    circuits[i].transform_qubits(lambda q: pair[q.x])
+                    for i, pair in zip(combination, layer_combinations.pairs)
+                )
+            )
+            zipped_circuits.append(
+                _ZippedCircuit(
+                    layer_i=layer_i,
+                    combination_i=combination_i,
+                    wide_circuit=wide_circuit,
+                    pairs=layer_combinations.pairs,
+                    combination=combination.tolist(),
+                )
+            )
+
+    # Construct truncated-with-measurement circuits to run!
     tasks = []
     for cycle_depth in cycle_depths:
-        for circuit_i, circuit in enumerate(circuits):
+        for zipped_circuit in zipped_circuits:
             circuit_depth = cycle_depth * 2 + 1
-            assert circuit_depth <= len(circuit)
-            truncated_circuit = circuit[:circuit_depth]
-            prepared_circuit = truncated_circuit + ops.measure(q0, q1)
+            assert circuit_depth <= len(zipped_circuit.wide_circuit)
+            # Slicing creates a copy, although this isn't documented
```

Comment:
Is this something you want to rely for future compatiblity?

Reply:
yes, although I would hope if anyone ever tries to change this in cirq it would cause a lot of tests to fail, be recognized as a breaking change, and be rejected. What do you think?

Reply:
I guess that is what would happen, it's fine I think ;)